knodle/trainer/cleanlab/latent_estimation.py

```diff
@@ -0,0 +1,64 @@
+import copy
+
+import numpy as np
+from sklearn.base import RegressorMixin
+from tqdm import tqdm
+
+from knodle.trainer.crossweigh_weighing.data_splitting_by_rules import (
+    k_folds_splitting_by_rules, k_folds_splitting_by_signatures
+)
+
+
+def estimate_cv_predicted_probabilities_split_by_rules(
+        model_input_x: np.ndarray,
+        labels: np.ndarray,
+        rule_matches_z: np.ndarray,
+        model: RegressorMixin,
+        num_classes: int,
+        cv_n_folds: int = 5,
+        seed: int = None,
+        other_class_id: int = None
+):
+    cv_train_datasets, cv_holdout_datasets = k_folds_splitting_by_rules(
+        model_input_x, labels, rule_matches_z, partitions=1, folds=cv_n_folds, seed=seed, other_class_id=other_class_id
+    )
+
+    return compute_psx_matrix(model, cv_train_datasets, cv_holdout_datasets, labels, num_classes)
+
+
+def estimate_cv_predicted_probabilities_split_by_signatures(
+        model_input_x: np.ndarray,
+        labels: np.ndarray,
+        rule_matches_z: np.ndarray,
+        model: RegressorMixin,
+        num_classes: int,
+        cv_n_folds: int = 5,
+        seed: int = None,
+        other_class_id: int = None
+):
+
+    cv_train_datasets, cv_holdout_datasets = k_folds_splitting_by_signatures(
+        model_input_x, labels, rule_matches_z, partitions=1, folds=cv_n_folds, seed=seed, other_class_id=other_class_id
+    )
```

Comment:
More of a refactoring question:
Does this have to output datasets, not e.g. tuples? Because later in `compute_psx_matrix` we unpack the dataset again, so this seems like an unneccessary step to create the TensorDatasets in the first place.
If CW expects a dataset output, that is ok then. But we could eventually consider to move dataset-wrapping to CW or a separate function, to avoid the packing-unpacking here for CL.

Reply:
I can do it, but then there would be 5 separate lists (train&test samples, train&test labels, test ids), and I thought it would be not optimal. or?